optuna/samplers/_gp/sampler.py

```diff
@@ -186,16 +286,33 @@ def sample_relative(
         )
         self._kernel_params_cache = kernel_params
 
+        # TODO(kAIto47802): If is_all_infeasible, the acquisition function for the objective
+        # function is ignored, so skipping the computation of kernel_params and acqf_params
+        # can improve speed.
         acqf_params = acqf.create_acqf_params(
             acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
             kernel_params=kernel_params,
             search_space=internal_search_space,
             X=normalized_params,
-            Y=standarized_score_vals,
+            Y=standardized_score_vals,
+            max_Y=-np.inf if is_all_infeasible else np.max(standardized_score_vals[is_feasible]),
+        )
+
+        constraints_acqf_params = self._get_constraints_acqf_params(
+            constraint_vals, internal_search_space, normalized_params
+        )
+
+        constrained_acqf_params = acqf.ConstrainedAcquisitionFunctionParams.from_acqf_params(
+            acqf_params, constraints_acqf_params
         )
 
         normalized_param = self._optimize_acqf(
-            acqf_params, normalized_params[np.argmax(standarized_score_vals), :]
+            constrained_acqf_params,
+            (
+                None
+                if is_all_infeasible
+                else normalized_params[np.argmax(standardized_score_vals[is_feasible]), :]
+            ),
```

Comment:
How about using an if statement like the following to make it clear that the logic remains unchanged in case constraints_func is None?

```python
        if self._constraints_func is not None:
            standardized_score_vals = standardized_score_vals[is_feasible]

        acqf_params = acqf.create_acqf_params(
            acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
            kernel_params=kernel_params,
            search_space=internal_search_space,
            X=normalized_params,
            Y=standardized_score_vals,
            max_Y=-np.inf if is_all_infeasible else np.max(standardized_score_vals[is_feasible]),
        )

        ...

        normalized_param = self._optimize_acqf(
            constrained_acqf_params,
            (
                None
                if is_all_infeasible
                else normalized_params[np.argmax(standardized_score_vals), :]
            ),
        )
```